source/opt/add_var_for_funccall_param.cpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/opt/add_var_for_funccall_param.h"
+
+#include "ir_builder.h"
+
+using namespace spvtools;
+using namespace opt;
+
+Pass::Status AddVarsForFuncCallParamPass::Process() {
+  auto result = Status::SuccessWithoutChange;
+  auto funcsNum = get_module()->end() - get_module()->begin();
+  if (funcsNum == 1) return result;
+  for (auto& func : *get_module()) {
+    // Get Variable insertion point
+    Instruction* varInsertPt = &*(func.begin()->begin());
+    func.ForEachInst([this, &result, &varInsertPt](Instruction* inst) {
+      if (inst->opcode() == SpvOpFunctionCall) {
+        Instruction* nextInst = inst->NextNode();
+        for (uint32_t i = 0; i < inst->NumInOperands(); ++i) {
+          auto& op = inst->GetInOperand(i);
+          if (op.type != SPV_OPERAND_TYPE_ID) continue;
+          Instruction* op_inst = get_def_use_mgr()->GetDef(op.AsId());
```

Comment:
`op_inst` is a unclear name ..
I recommend setting it as `operand_inst` and creating a separate method for the following lines `InstructionBuilder builder(context(), inst, .. `. Since `if (op_inst->opcode() == SpvOpAccessChain) {` checks the opcode, we can set the parameter name for the instruction as `access_chain` in the separate method, which would be more clear.

In addition, I recommend `ReplaceAccessChainFunctionCallArgument` for the separate method name.